### PR TITLE
CORE-12423: Adjust aprove registration to keep the member status

### PR DESCRIPTION
**Pull request by yift-r3**
Make sure that a suspended member status will not get overwritten.

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]

File: components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/UpdateMemberAndRegistrationRequestToApprovedHandler.kt [hunk 1]
```diff
@@ -106,4 +99,22 @@ internal class UpdateMemberAndRegistrationRequestToApprovedHandler(
             )
         }
     }
+
+    private fun processMgmContext(
```
Comment (dimosr) [inline components/membership/membership-persistence-service-impl/src/main/kotlin/net/corda/membership/impl/persistence/service/handler/UpdateMemberAndRegistrationRequestToApprovedHandler.kt]: isn't the status at this point always going to be `PENDING` given we query for the pending record (effectively always switching it to `ACTIVE`)?

File: components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/UpdateMemberAndRegistrationRequestToApprovedHandlerTest.kt [hunk 2]
```diff
@@ -229,6 +230,90 @@ class UpdateMemberAndRegistrationRequestToApprovedHandlerTest {
         )
     }
 
+    @Test
+    fun `invoke will keep the state if it was suspended`() {
```
Comment (dimosr) [inline components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/UpdateMemberAndRegistrationRequestToApprovedHandlerTest.kt:234]: Do we have a test that exercises the scenario where no entity is returned for the query that searches for non-pending entries (ACTIVE or SUSPENDED)?

**Review by dimosr — COMMENTED** [review]
Logic looks good to me now, just a comment on the tests

File: components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/UpdateMemberAndRegistrationRequestToApprovedHandlerTest.kt (hunk 2 above)
Comment (yift-r3) [reply components/membership/membership-persistence-service-impl/src/test/kotlin/net/corda/membership/impl/persistence/service/handler/UpdateMemberAndRegistrationRequestToApprovedHandlerTest.kt:234]: In the test above it (`invoke will update the MGM context members`) the search is returning null

**Review by dimosr — APPROVED** [review]

**Pull request merged**